### PR TITLE
Clean up clippy warnings/errors

### LINKED ISSUE #99: Clean up clippy warnings and errors
Right now the CI check `Rust / Lints` always fails on historical code due to clippy complaining. In order to expunge those errors from the CI check and allow meaningful use of clippy in development, those warnings should be silenced or the code should be corrected.

### PR DESCRIPTION
**Pull request by emostov**
Closes #99, see issue for fix description.

### CODE REVIEW

File: src/actors.rs [hunk 1]
```diff
@@ -298,7 +298,7 @@ where
             .collect();
         let difference: HashSet<u32> = missing_storage_nums
             .difference(&blocks)
-            .map(|b| *b)
+            .copied()
```
Comment (dvdplm) [inline src/actors.rs:301]: This change is good. I wonder however if we couldn't combine `missing_storage_nums` and this pile of combinators into one, thus avoiding having to allocate for `missing_storage_nums`?

File: src/actors/workers/database.rs [hunk 2]
```diff
@@ -96,7 +96,7 @@ impl<B: BlockT> DatabaseActor<B> {
     async fn batch_storage_handler(&self, storage: Vec<Storage<B>>) -> Result<()> {
         let mut conn = self.db.conn().await?;
         let mut block_nums: Vec<u32> = storage.iter().map(|s| s.block_num()).collect();
-        block_nums.sort();
+        block_nums.sort_unstable();
```
Comment (dvdplm) [inline src/actors/workers/database.rs:99]: What did clippy complain about here?

File: src/backend/util.rs [hunk 3]
```diff
@@ -141,9 +141,9 @@ pub fn read_db<Block>(
 where
     Block: BlockT,
 {
-    block_id_to_lookup_key(db, col_index, id).and_then(|key| match key {
-        Some(key) => Ok(db.get(col, key.as_ref())),
-        None => Ok(None),
+    block_id_to_lookup_key(db, col_index, id).map(|key| match key {
+        Some(key) => db.get(col, key.as_ref()),
+        None => None,
```
Comment (dvdplm) [inline src/backend/util.rs:146]: Would this work?

```
    block_id_to_lookup_key(db, col_index, id)
        .map(|key|
            key.and_then(|key| db.get(col, key.as_ref()))
        )
```

If it does I'm still not sure I like it better. Explicit is good sometimes.

**Review by dvdplm — APPROVED** [review]
lgtm modulo nits

File: src/actors/workers/database.rs (hunk 2 above)
Comment (insipx) [reply src/actors/workers/database.rs:99]: I think it's this lint: https://github.com/rust-lang/rust-clippy/issues/5762

**Review by insipx — APPROVED** [review]

File: src/actors/workers/database.rs (hunk 2 above)
Comment (dvdplm) [reply src/actors/workers/database.rs:99]: I see, TIL. :)

File: src/actors.rs (hunk 1 above)
Comment (emostov) [reply src/actors.rs:300]: Ok, I just combined it into one, however I don't see how this avoids a non-constant allocation. From my understanding we would still have to allocate at `.collect::<HashSet<u32>>()`, which is the equivalent of `missing_storage_nums` minus the variable binding on the stack. But I am guessing I am missing something?

Comment (dvdplm) [reply src/actors.rs:300]: I don't know how what you have now optimizes, but before we had two `collect()` calls (which each allocates); with this last change I'd hope we only allocate once and for a smaller collection, i.e. just the difference.

**Review by dvdplm — APPROVED** [review]

**Pull request merged**